relayer/relayer.go

```diff
@@ -93,17 +68,38 @@ func (r *Relayer) Start(ctx context.Context) error {
 				return err
 			}
 
-			cmd := exec.Command("near", "call", r.daAccountId, "submit", "--base64", base64.StdEncoding.EncodeToString(encodedBlock), "--accountId", r.daAccountId)
-			cmd.Env = os.Environ()
-			out, err := cmd.CombinedOutput()
-
-			r.logger.Info(string(out))
+			out, err := r.nearClient.ForceSubmit(encodedBlock)
 			if err != nil {
-				r.logger.Errorf("error calling NEAR account: %s", err.Error())
+				r.logger.Error("Error submitting block to NEAR", "err", err)
 				return err
 			}
+
+			r.logger.Info(string(out))
 		case <-ctx.Done():
 			return nil
 		}
 	}
 }
+
+func (r *Relayer) getBlockByNumber(ctx context.Context, number *big.Int) *ethtypes.Block {
+	r.logger.Infof("Got rollup block: #%s", number.String())
+
+	for i := 0; i < 5; i++ {
+		select {
+		case <-ctx.Done():
+			return nil
+		default:
+		}
+
+		block, err := r.rpcClient.BlockByNumber(ctx, number)
+		if err != nil {
+			r.logger.Errorf("Error fetching rollup block: %s", err.Error())
+			time.Sleep(1 * time.Second)
+			continue
+		}
+
+		return block
+	}
+
+	panic("Could not fetch rollup block")
```

Comment:
Shouldn't this be a logger error or such?

Reply:
Ideally would return an error that would be logged later on

Reply:
That is a copy of an old implementation.

Reply:
Can be error

Reply:
done